api/controllers/StatsController.js

```diff
@@ -1,11 +1,40 @@
 const dayjs = require('dayjs');
-const isBetween = require('dayjs/plugin/isBetween');
 const Result = require('../../types/Result.es5');
-dayjs.extend(isBetween);
 
 /////////////
 // Helpers //
 /////////////
+/**
+ * Find the season in which a given timestamp occurred using binary search
+ * @param {*} sortedSeasons - Seasons sorted (OLDEST FIRST)
+ * @param {*} timeStamp - Desired time in millis since epoch
+ * @returns the season | null if none is found
+ */
+function getSeasonFromTimeStamp(sortedSeasons, timeStamp) {
+  if (sortedSeasons.length <= 0) {
+    return null;
+  }
+  let minIndex = 0;
+  let maxIndex = sortedSeasons.length - 1;
+  let currentIndex = Math.floor((minIndex + maxIndex) / 2);
+  // let currentSeason = sortedSeasons[currentIndex];
+  while (minIndex <= maxIndex) {
+    const currentSeason = sortedSeasons[currentIndex];
+    if (timeStamp >= currentSeason.startTime && timeStamp <= currentSeason.endTime) {
+      return currentSeason;
+    }
+    // Target time is before current season
+    if (timeStamp < currentSeason.startTime) {
+      minIndex = currentIndex + 1;
+    // Target time is after current season
+    } else {
+      maxIndex = currentIndex - 1;
+    }
+    currentIndex = currentIndex = Math.floor((minIndex + maxIndex) / 2);
+  }
+  return null;
+}
+
```

Comment:
```suggestion
function getSeasonFromTimeStamp(sortedSeasons, timeStamp) {
  if (sortedSeasons.length <= 0) {
    return null;
  }
  return sortedSeasons.find(
    ({ startTime, endTime }) => startTime <= timeStamp && timeStamp <= endTime
  );
}

```

Reply:
I'm not sure if there's a particular reason for using Binary search here, but I think the .find method could work.

Reply:
Okay, so It seems like binary search is faster than the .find method since it's already sorted. However I wonder if the performance outweighs the complexity/readability of the code

Reply:
This was originally a `.find()` call, but I refactored it to binary search for performance reasons. `.find()` is a linear search through the entire array, which takes up to n loops, where n is the array length. And since we’re looping through the matches (and later the games), using .find takes m*n loops at max ie O(n^2). 

Binary search cuts down the number of iterations because the assumption of having sorted input means we can check the halfway point and cut the list in half every time, making the max number of checks log_(n) giving the algorithm O(nlog_2(n)) which should be faster

Reply:
The performance vs readability trade off is a fair question and it may not be worth it to use binary search, but since this code loops through every game and every match in the site's history I think it makes sense to try to optimize it for speed

Reply:
This is the one time readability is less important :)